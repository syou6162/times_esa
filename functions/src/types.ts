/**
 * Firebase Functionsのリクエスト/レスポンス型定義
 */

import { type DailyReportCategory } from './dateUtils';

// 設定型定義
export type EsaConfig = {
  teamName: string;
  accessToken: string;
}

// リクエスト型定義
export type SubmitTextRequest = {
  category: DailyReportCategory;
  tags: string[];
  title: string;
  text: string;
}

export type DailyReportRequest = {
  category: DailyReportCategory;
}

export type RecentDailyReportsRequest = {
  days?: number; // 過去何日分を取得するか（デフォルト: 10）
}

// エラーレスポンス型定義
export type EsaErrorResponse = {
  error: string;
  message: string;
}