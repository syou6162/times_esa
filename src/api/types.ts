import type { HttpsCallableResult } from 'firebase/functions';

// リクエスト/レスポンスの型定義
export type DailyReportRequest = {
  category: string;
}

export type DailyReportResponse = {
  updatedAt: string;
  url: string;
  bodyMd: string;
  bodyHtml: string;
  tags: string[];
  name: string;
  category: string;
}

export type TagListRequest = {}

export type TagListResponse = {
  tags: Array<{
    name: string;
    postsCount: number;
  }>;
  totalCount: number;
}

export type SubmitTextToEsaRequest = {
  category: string;
  tags: string[];
  title: string;
  text: string;
}

export type SubmitTextToEsaResponse = {
  updatedAt: string;
  url: string;
  bodyMd: string;
  bodyHtml: string;
  tags: string[];
  name: string;
  category: string;
}

export type RecentDailyReportsRequest = {}

export type RecentDailyReportsResponse = {
  reports: Array<{
    date: string;
    url: string;
    posts_count: number;
  }>;
}

// APIクライアントのインターフェース
export interface ApiClient {
  getDailyReport(category: string): Promise<HttpsCallableResult<DailyReportResponse>>;
  getTagList(): Promise<HttpsCallableResult<TagListResponse>>;
  submitTextToEsa(
    category: string,
    tags: string[],
    title: string,
    text: string
  ): Promise<HttpsCallableResult<SubmitTextToEsaResponse>>;
  getRecentDailyReports(): Promise<HttpsCallableResult<RecentDailyReportsResponse>>;
}