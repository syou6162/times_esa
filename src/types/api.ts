/**
 * API関連の型定義
 * 
 * APIのリクエスト・レスポンスに関する型を定義
 * 基本的にAPIの仕様に合わせてsnake_caseを使用
 */

// 日報取得API
export type DailyReportRequestType = {
  category: string;
};

export type DailyReportResponseType = {
  updated_at: string;
  url: string;
  body_md: string;
  body_html: string;
  tags: string[];
  name: string;
  category: string;
};

// タグ一覧取得API
export type TagListRequestType = Record<string, never>;

export type TagListResponseType = {
  tags: Array<{
    name: string;
    posts_count: number;
  }>;
};

// 最近の日報一覧取得API
export type RecentDailyReportsRequest = {
  days?: number; // 過去何日分を取得するか（デフォルト: 10）
};

export type RecentDailyReportsResponse = {
  reports: Array<{
    date: string;
    title: string;
    posts_count: number;
    tags: string[];
    formatted_date: string;
    url?: string;
  }>;
  total_count: number;
};

// ESA投稿API
export type SubmitTextToEsaRequestType = {
  category: string;
  tags: string[];
  title: string;
  text: string;
};

export type SubmitTextToEsaResponseType = {
  updated_at: string;
  url: string;
  body_md: string;
  body_html: string;
  tags: string[];
  name: string;
  category: string;
};