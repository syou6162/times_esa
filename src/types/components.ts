// コンポーネントのProps型定義をまとめたファイル

import { Auth } from 'firebase/auth';
import { AutocompleteProps } from '@mui/material/Autocomplete';
import { GoogleUser, InputChangeHandler, TextAreaChangeHandler } from './index';

// Body Component Props
export type BodyProps = {
  hasUserLanded: boolean;
  isSignedIn: boolean;
  user: GoogleUser;
  firebaseAuth: Auth | null;
}

// DailyReport Component Props
export type DailyReportProps = {
  fetching: boolean;
  fetchErrorMessage: string;
  esaText: string;
  esaHtml: string;
  reloadDailyReport: () => void;
}

// DailyReport内部のコンポーネントProps
export type DailyReportHtmlProps = {
  esaHtml: string;
}

export type DailyReportTextProps = {
  esaText: string;
}

export type DailyReportShareProps = {
  esaText: string;
}

// ShareButton Props
export type TweetButtonProps = {
  text: string;
}

export type CopyButtonProps = {
  text: string;
}

// EsaSubmitForm Props
export type EsaSubmitFormProps = {
  category: string;
  title: string;
  tags: string[];
  tagCandidates: string[];
  fetching: boolean;
  onSubmit: (
    category: string,
    title: string,
    markdown: string,
    html: string,
    tags: string[]
  ) => void;
}

// EsaTitleField Props
export type EsaTitleFieldProps = {
  fetching: boolean;
  sending: boolean;
  title: string;
  onChange: InputChangeHandler;
}

// EsaTextField Props
export type EsaTextFieldProps = {
  sending: boolean;
  text: string;
  onChange: TextAreaChangeHandler;
}

// Autocomplete用の型エイリアス
type AutocompleteChangeHandler<T> = AutocompleteProps<T, true, false, true>['onChange'];

// EsaTagsField Props
export type EsaTagsFieldProps = {
  fetching: boolean;
  sending: boolean;
  tags: string[];
  tagCandidates: string[];
  onChange: AutocompleteChangeHandler<string>;
}

// TimesEsa Props
export type TimesEsaProps = {
  canFetchCloudFunctionEndpoints: boolean;
}

// DailyReportsList Props
export type DailyReportsListProps = {
  selectedDate?: string;
  onDateSelect: (date: string) => void;
}

// SignInDialog Props
export type SignInDialogProps = {
  firebaseAuth: Auth | null;
}

// SignOutButton Props
export type SignOutButtonProps = {
  firebaseAuth: Auth | null;
}

// StyledFirebaseAuth Props
export type StyledFirebaseAuthProps = {
  firebaseAuth: Auth | null;
}

// Footer Props
export type FooterProps = {
  isSignedIn: boolean;
  user: GoogleUser;
  firebaseAuth: Auth | null;
}