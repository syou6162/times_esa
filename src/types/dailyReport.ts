/**
 * 日報関連の型定義
 */

export type Tag = {
  name: string;
  posts_count: number; // eslint-disable-line camelcase
};

export type DailyReportSummary = {
  date: string; // yyyy-MM-dd形式
  title: string;
  postsCount: number;
  tags: string[];
  formattedDate: string; // 日本語日付（例: "6月19日(水)"）
  url?: string; // esaのURL
};

export type DailyReportContent = {
  updated_at: string;
  url: string;
  body_md: string;
  body_html: string;
  tags: string[];
  name: string;
  category: string;
};

export type DailyReportRequestType = {
  category: string;
};

export type DailyReportResponseType = DailyReportContent;

export type TagListRequestType = Record<string, never>;

export type TagListResponseType = {
  tags: Tag[];
};

export type RecentDailyReportsRequest = {
  days?: number; // 過去何日分を取得するか（デフォルト: 10）
};

export type RecentDailyReportsResponse = {
  reports: DailyReportSummary[];
  total_count: number;
};

/**
 * 日報の表示モード
 */
export type ReportViewMode = 'today' | 'past';

/**
 * 日報選択の状態
 */
export type SelectedDateState = 'today' | string; // 'today' または yyyy-MM-dd形式