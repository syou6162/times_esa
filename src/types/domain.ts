/**
 * ドメインモデルの型定義
 * 
 * アプリケーション内で使用するドメインモデルの型を定義
 * 基本的にcamelCaseを使用
 */

// タグ情報
export type Tag = {
  name: string;
  posts_count: number; // eslint-disable-line camelcase
};

// 日報サマリー（一覧表示用）
export type DailyReportSummary = {
  date: string; // yyyy-MM-dd形式
  title: string;
  postsCount: number;
  tags: string[];
  formattedDate: string; // 日本語日付（例: "6月19日(水)"）
  url?: string; // esaのURL
};

// 日報コンテンツ（API responseの形式）
export type DailyReportContent = {
  updated_at: string;
  url: string;
  body_md: string;
  body_html: string;
  tags: string[];
  name: string;
  category: string;
};

// 日報データ（アプリ内部で使用する形式、camelCase）
export type DailyReportData = {
  updatedAt: string;
  url: string;
  text: string;      // body_md → text
  html: string;      // body_html → html
  tags: string[];
  title: string;     // name → title
  category: string;
};

// ユーザー情報
export type GoogleUser = {
  email: string;
  displayName: string;
  photoURL: string;
};

// 日報の表示モード
export type ReportViewMode = 'today' | 'past';

// 日報選択の状態
export type SelectedDateState = 'today' | string; // 'today' または yyyy-MM-dd形式