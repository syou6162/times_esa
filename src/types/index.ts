// 共通の型定義をまとめたファイル

// ユーザー情報の型定義
export type GoogleUser = {
  email: string;
  displayName: string;
  photoURL: string;
}

// タグの型定義（バックエンドのEsaTagCamelCaseと同じ構造）
export type Tag = {
  name: string;
  postsCount: number;
}

// 日報表示タイプのenum
export enum DailyReportType {
  HTML = 'HTML',
  TEXT = 'TEXT',
  SHARE = 'SHARE',
}

// Form要素のイベントハンドラー型
export type InputChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => void;
export type TextAreaChangeHandler = (e: React.ChangeEvent<HTMLTextAreaElement>) => void;
