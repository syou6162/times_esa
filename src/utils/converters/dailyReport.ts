/**
 * 日報データの変換関数
 * 
 * ESA APIのレスポンス（snake_case）をアプリ内部のモデル（camelCase）に変換
 */

import { 
  DailyReportResponseType, 
  TagListResponseType,
  RecentDailyReportsResponse 
} from '../../types/api';
import { 
  DailyReportData, 
  Tag,
  DailyReportSummary 
} from '../../types/domain';

/**
 * 日報レスポンスをアプリ内部のデータ形式に変換
 */
export const convertDailyReportResponse = (response: DailyReportResponseType): DailyReportData => {
  return {
    updatedAt: response.updated_at,
    url: response.url,
    text: response.body_md,
    html: response.body_html,
    tags: response.tags,
    title: response.name,
    category: response.category,
  };
};

/**
 * タグレスポンスをアプリ内部のデータ形式に変換
 * 
 * 現状はそのまま返すだけだが、将来的な変更に備えて変換関数を用意
 */
export const convertTagListResponse = (response: TagListResponseType): Tag[] => {
  return response.tags;
};

/**
 * 最近の日報一覧レスポンスをアプリ内部のデータ形式に変換
 */
export const convertRecentReportsResponse = (response: RecentDailyReportsResponse): DailyReportSummary[] => {
  return response.reports.map(report => ({
    date: report.date,
    title: report.title,
    postsCount: report.posts_count,  // snake_case → camelCase
    tags: report.tags,
    formattedDate: report.formatted_date,  // snake_case → camelCase
    url: report.url,
  }));
};