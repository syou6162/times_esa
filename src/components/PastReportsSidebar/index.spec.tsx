// @vitest-environment jsdom
import { fireEvent, render } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { PastReportsSidebar, PastReportsSidebarProps } from '.';
import { DailyReportSummary } from '../../types/dailyReport';

const mockReports: DailyReportSummary[] = [
  {
    date: '2024-06-19',
    title: '開発、会議、レビュー',
    postsCount: 5,
    tags: ['開発', 'times_esa'],
    formattedDate: '6月19日(水)'
  },
  {
    date: '2024-06-18',
    title: 'レビュー、設計',
    postsCount: 3,
    tags: ['設計', 'レビュー'],
    formattedDate: '6月18日(火)'
  },
];

describe('PastReportsSidebarが正しく表示される', () => {
  it('今日の日報が選択された状態で表示される', () => {
    const mockOnSelectReport = vi.fn();
    const props: PastReportsSidebarProps = {
      reports: mockReports,
      selectedDate: 'today',
      onSelectReport: mockOnSelectReport,
    };

    const renderResult = render(<PastReportsSidebar {...props} />);

    expect(renderResult.asFragment()).toMatchSnapshot();
    expect(renderResult.getByText('今日の日報')).toBeTruthy();
    expect(renderResult.getByText('編集可能')).toBeTruthy();
    expect(renderResult.getByText('6月19日(水)')).toBeTruthy();
    expect(renderResult.getByText('開発、会議、レビュー')).toBeTruthy();
  });

  it('過去の日報が選択された状態で表示される', () => {
    const mockOnSelectReport = vi.fn();
    const props: PastReportsSidebarProps = {
      reports: mockReports,
      selectedDate: '2024-06-19',
      onSelectReport: mockOnSelectReport,
    };

    const renderResult = render(<PastReportsSidebar {...props} />);

    expect(renderResult.asFragment()).toMatchSnapshot();
    expect(renderResult.getByText('開発、会議、レビュー')).toBeTruthy();
    expect(renderResult.getByText('5個のつぶやき')).toBeTruthy();
    expect(renderResult.getByText('開発')).toBeTruthy();
    expect(renderResult.getByText('times_esa')).toBeTruthy();
  });

  it('今日の日報をクリックするとコールバックが呼ばれる', () => {
    const mockOnSelectReport = vi.fn();
    const props: PastReportsSidebarProps = {
      reports: mockReports,
      selectedDate: '2024-06-19',
      onSelectReport: mockOnSelectReport,
    };

    const renderResult = render(<PastReportsSidebar {...props} />);

    fireEvent.click(renderResult.getByText('今日の日報'));
    expect(mockOnSelectReport).toHaveBeenCalledWith('today');
  });

  it('過去の日報をクリックするとコールバックが呼ばれる', () => {
    const mockOnSelectReport = vi.fn();
    const props: PastReportsSidebarProps = {
      reports: mockReports,
      selectedDate: 'today',
      onSelectReport: mockOnSelectReport,
    };

    const renderResult = render(<PastReportsSidebar {...props} />);

    fireEvent.click(renderResult.getByText('開発、会議、レビュー'));
    expect(mockOnSelectReport).toHaveBeenCalledWith('2024-06-19');
  });

  it('空の日報リストでも正しく表示される', () => {
    const mockOnSelectReport = vi.fn();
    const props: PastReportsSidebarProps = {
      reports: [],
      selectedDate: 'today',
      onSelectReport: mockOnSelectReport,
    };

    const renderResult = render(<PastReportsSidebar {...props} />);

    expect(renderResult.asFragment()).toMatchSnapshot();
    expect(renderResult.getByText('日報一覧')).toBeTruthy();
    expect(renderResult.getByText('今日の日報')).toBeTruthy();
  });
});