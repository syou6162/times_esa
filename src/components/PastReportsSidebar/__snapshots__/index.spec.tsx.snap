// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PastReportsSidebarが正しく表示される > 今日の日報が選択された状態で表示される 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kak0am"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-31h6qn-MuiTypography-root"
    >
      日報一覧
    </h6>
    <ul
      class="MuiList-root MuiList-padding css-13rb8qi-MuiList-root"
    >
      <button
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1nbyyxq-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-64nssn-MuiTypography-root"
          >
            今日の日報
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-3k0zn9-MuiTypography-root"
          >
            編集可能
          </p>
        </div>
      </button>
      <button
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-wjrk7s-MuiListItem-root"
      >
        <div
          class="MuiBox-root css-8atqhb"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ii4reh-MuiTypography-root"
          >
            6月19日(水)
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1oq2vc5-MuiTypography-root"
          >
            開発、会議、レビュー
          </p>
          <div
            class="MuiBox-root css-107ld1r"
          >
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-q1fxe1-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
              >
                開発
              </span>
            </div>
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-q1fxe1-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
              >
                times_esa
              </span>
            </div>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption css-1rrr8pd-MuiTypography-root"
          >
            5個のつぶやき
          </span>
        </div>
      </button>
      <button
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-wjrk7s-MuiListItem-root"
      >
        <div
          class="MuiBox-root css-8atqhb"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ii4reh-MuiTypography-root"
          >
            6月18日(火)
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1oq2vc5-MuiTypography-root"
          >
            レビュー、設計
          </p>
          <div
            class="MuiBox-root css-107ld1r"
          >
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-q1fxe1-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
              >
                設計
              </span>
            </div>
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-q1fxe1-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
              >
                レビュー
              </span>
            </div>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption css-1rrr8pd-MuiTypography-root"
          >
            3個のつぶやき
          </span>
        </div>
      </button>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`PastReportsSidebarが正しく表示される > 空の日報リストでも正しく表示される 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kak0am"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-31h6qn-MuiTypography-root"
    >
      日報一覧
    </h6>
    <ul
      class="MuiList-root MuiList-padding css-13rb8qi-MuiList-root"
    >
      <button
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1nbyyxq-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-64nssn-MuiTypography-root"
          >
            今日の日報
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-3k0zn9-MuiTypography-root"
          >
            編集可能
          </p>
        </div>
      </button>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`PastReportsSidebarが正しく表示される > 過去の日報が選択された状態で表示される 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kak0am"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-31h6qn-MuiTypography-root"
    >
      日報一覧
    </h6>
    <ul
      class="MuiList-root MuiList-padding css-13rb8qi-MuiList-root"
    >
      <button
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-58bp3r-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-plhci5-MuiTypography-root"
          >
            今日の日報
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-3k0zn9-MuiTypography-root"
          >
            編集可能
          </p>
        </div>
      </button>
      <button
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-q8c0aq-MuiListItem-root"
      >
        <div
          class="MuiBox-root css-8atqhb"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-spsapl-MuiTypography-root"
          >
            6月19日(水)
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1oq2vc5-MuiTypography-root"
          >
            開発、会議、レビュー
          </p>
          <div
            class="MuiBox-root css-107ld1r"
          >
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-q1fxe1-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
              >
                開発
              </span>
            </div>
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-q1fxe1-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
              >
                times_esa
              </span>
            </div>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption css-1rrr8pd-MuiTypography-root"
          >
            5個のつぶやき
          </span>
        </div>
      </button>
      <button
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-wjrk7s-MuiListItem-root"
      >
        <div
          class="MuiBox-root css-8atqhb"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ii4reh-MuiTypography-root"
          >
            6月18日(火)
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1oq2vc5-MuiTypography-root"
          >
            レビュー、設計
          </p>
          <div
            class="MuiBox-root css-107ld1r"
          >
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-q1fxe1-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
              >
                設計
              </span>
            </div>
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-q1fxe1-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
              >
                レビュー
              </span>
            </div>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption css-1rrr8pd-MuiTypography-root"
          >
            3個のつぶやき
          </span>
        </div>
      </button>
    </ul>
  </div>
</DocumentFragment>
`;
